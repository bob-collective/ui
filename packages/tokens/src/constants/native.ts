import { ChainId } from '@gobob/chains';
import { ERC20Token } from '@gobob/currency';

import { WETH9 } from './weth';

export const WNATIVE = {
  [ChainId.ETHEREUM]: WETH9[ChainId.ETHEREUM],
  [ChainId.SEPOLIA]: WETH9[ChainId.SEPOLIA],
  [ChainId.BOB]: '' as any,
  [ChainId.BOB_SEPOLIA]: '' as any,
  [ChainId.ARBITRUM_ONE]: '' as any,
  [ChainId.BASE]: '' as any,
  [ChainId.OP]: '' as any,
  [ChainId.BSC]: '' as any,
  [ChainId.OPBNB]: '' as any,
  [ChainId.POLYGON]: '' as any,
  [ChainId.POLYGON_ZKEVM]: '' as any,
  [ChainId.MOONBEAM]: '' as any
} satisfies Record<ChainId, ERC20Token>;

const ETHER = { name: 'Ether', symbol: 'ETH', decimals: 18 } as const;

export const NATIVE = {
  [ChainId.ETHEREUM]: ETHER,
  [ChainId.SEPOLIA]: {
    name: 'Sepolia Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.BOB]: {
    name: 'BOB Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.BOB_SEPOLIA]: {
    name: 'BOB Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.ARBITRUM_ONE]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.BASE]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.OP]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.BSC]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.OPBNB]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.POLYGON]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.POLYGON_ZKEVM]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  },
  [ChainId.MOONBEAM]: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
} satisfies Record<
  ChainId,
  {
    name: string;
    symbol: string;
    decimals: number;
  }
>;
