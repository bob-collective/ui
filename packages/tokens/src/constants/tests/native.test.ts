import { describe, it, expect } from 'vitest';

import { WNATIVE, NATIVE } from '../native';

describe('WNATIVE and NATIVE Token Snapshots', () => {
  it('should match snapshot for WNATIVE', () => {
    expect(WNATIVE).toMatchSnapshot();
  });

  it('should match snapshot for NATIVE', () => {
    expect(NATIVE).toMatchSnapshot();
  });
});
