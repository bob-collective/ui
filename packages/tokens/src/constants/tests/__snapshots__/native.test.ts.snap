// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WNATIVE and NATIVE Token Snapshots > should match snapshot for NATIVE 1`] = `
{
  "1": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "10": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "1101": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "111": {
    "decimals": 18,
    "name": "BOB Ether",
    "symbol": "ETH",
  },
  "11155111": {
    "decimals": 18,
    "name": "Sepolia Ether",
    "symbol": "ETH",
  },
  "1284": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "137": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "200901": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "204": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "4200": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "42161": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "56": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
  "60808": {
    "decimals": 18,
    "name": "BOB Ether",
    "symbol": "ETH",
  },
  "808813": {
    "decimals": 18,
    "name": "BOB Ether",
    "symbol": "ETH",
  },
  "8453": {
    "decimals": 18,
    "name": "Ether",
    "symbol": "ETH",
  },
}
`;

exports[`WNATIVE and NATIVE Token Snapshots > should match snapshot for WNATIVE 1`] = `
{
  "1": ERC20Token {
    "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
    "chainId": 1,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
  "10": "",
  "1101": "",
  "111": "",
  "11155111": ERC20Token {
    "address": "0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9",
    "chainId": 11155111,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
  "1284": "",
  "137": "",
  "200901": "",
  "204": "",
  "4200": "",
  "42161": "",
  "56": "",
  "60808": "",
  "808813": "",
  "8453": "",
}
`;
